### ISSUE TITLE
Enhancement: no-commented-out-code aggressive mode toggle

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#31]
## Description

The `no-commented-out-code` rule can be overly aggressive, flagging comments that look like code but are intentionally commented for documentation or examples.

## Example False Positives

```ts
// Usage example:
// const client = new Client({ apiKey: 'xxx' });
// await client.connect();
```

This would be flagged as commented-out code.

## Suggested Enhancement

Add an option to control aggressiveness:

```ts
{
  "llm-core/no-commented-out-code": ["error", {
    "mode": "strict" | "moderate" | "lenient"
  }]
}
```

- **strict**: Current behavior
- **moderate**: Require 2+ lines of consecutive code-like comments
- **lenient**: Require 3+ lines OR block comments only

## Alternative

Add `allowExamples: true` option that looks for markers like `// Example:` or `// Usage:` above code blocks.
